### DISCUSSION TITLE
Node for selecting and coloring a specific sequence of a virus

### DISCUSSION THREAD

**Discussion by pgb3** [BradyAJohnston/MolecularNodes#652] [General]
Thank you very much for your wonderful addon and your dedication to make it excellent.
I imported a virus with many subunits into Blender. I would like to color a specific sequence in each subunit with a solid color. This can be easily done in QuimeraX, but in Molecular Nodes I did not find a node in which to introduce a peptide sequence. I found a "select residue ID range" node but since this virus has about 60 subunits it would be very tedious to find out the residue ID of this sequence in each of the 60 subunits.
I am not very familiar with Molecular Nodes. Maybe there is a way to do it, but if not it would be a great enhancement to your software.

**Comment by BradyAJohnston** [comment]
I am not 100% sure about what you are after. Are you wanting to select the same amino acid sequence in each of the chains? So the same amino acids, but different chains? Are the chains themselves the same protein? Because if so then the Res ID Range will work

**Comment by pgb3** [comment]
Thank you very much for your prompt reply. All ~60 subunits are the same 527 amino acids forming the virus capsid. I want to select the same 10 amino acid sequence in each of the chains to give them a different color. I know the amino acid sequence but I do not know the ID numbers. It would be great if I could select the amino acid sequence directly like I did in QuimeraX.
After reading your reply and looking into the cif file in more detail, I realized I could search for this 10 aa sequence in the first subunit of the cif file to find out the corresponding IDs and then use the Res ID Range node in Blender to color them differently. This will certainly help me and I am very grateful you suggested it. Still I little bit more cumbersome than having a node that reads and selects the aa sequence.
I will try Res ID Range and hopefully I can make it work. Thank you so much.

**Reply by BradyAJohnston** [reply]
This is currently a limitation of Geometry Nodes. Nodes don't support text as an attribute, so it's not possible to create a node that takes a string as an input (AA sequence) and creates a selection. Just a limit of the system we are working in. 

But yes as you have devised, you can use the _Res ID Range_ and that will give you a selection for each chain in the protein
